MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,68 @@
+﻿using AutoMapper;
+using MeetupPlatformApi.Authentification;
+using MeetupPlatformApi.Context;
+using MeetupPlatformApi.DataTransferObjects;
+using MeetupPlatformApi.Entities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Security.Claims;
+
+namespace MeetupPlatformApi.Controllers
+{
+    [Route("/api/meetups/authentication")]
+    [ApiController]
+    public class AuthenticationController : ControllerBase
+    {
+        private readonly ApplicationContext context;
+        private readonly IMapper mapper;
+        private readonly AuthentificationManager authentificationManager;
+
+        public AuthenticationController(ApplicationContext context, IMapper mapper, AuthentificationManager authentificationManager)
+        {
+            this.context = context;
+            this.mapper = mapper;
+            this.authentificationManager = authentificationManager;
+        }
+
+        [HttpGet("{id}", Name = "UserById")]
+        public async Task<IActionResult> GetUserById(Guid id)
+        {
+            var user = await context.Users.SingleOrDefaultAsync(u => u.Id.Equals(id));
+            var userOutput = mapper.Map<UserOutputDto>(user);
+
+            return userOutput == null ? Ok(userOutput) : NotFound();
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> RegisterUser([FromBody] UserForCreationDto userFromBody)
+        {
+            var user = mapper.Map<UserEntity>(userFromBody);
+
+            await context.Users.AddAsync(user);
+            await context.SaveChangesAsync();
+
+            return CreatedAtRoute("UserById", new { id = user.Id }, user);
+        }
```

Comment:
it seems to me that when registering, you should give the path not to GetUserById, but to Login, so that the user can go into the system, that is, he does not need to know his personal information during registration, he needs to find out how to get into the system, that is, get a token

Reply:
1. `CreateAtRoute` was originally designed for CRUD scenarios, where we can provide url for the newly created resource (that's actually a part of the HATEOAS approach), so we **definitely should** provide a url for the GET user action (in this case by id)
2. The **wrong** approach was chosen for specifying the name of the action - You need to use the `nameof` keyword (so that if the method is renamed, the reference to the action is also updated).